### PR TITLE
implement vectorized evaluation for `builtinConvSig`

### PR DESCRIPTION
**Pull request by ZhaoQian888**
PCP #12106 
### What problem does this PR solve? 
implement vectorized evaluation for `builtinConvSig`

### What is changed and how it works?

```
BenchmarkVectorizedBuiltinMathFunc/builtinConvSig-VecBuiltinFunc-8                172767              6451 ns/op               0 B/op          0 allocs/op

BenchmarkVectorizedBuiltinMathFunc/builtinConvSig-NonVecBuiltinFunc-8              36088             33235 ns/op               0 B/op          0 allocs/op
```
### Check List 

Tests 

 - Unit test
 -

### CODE REVIEW

File: expression/builtin_math_vec.go [hunk 1]
```diff
@@ -1049,11 +1049,101 @@ func (b *builtinSignSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) er
...
 
 func (b *builtinConvSig) vecEvalString(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf1, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf1)
+	buf2, err := b.bufAllocator.get(types.ETString, n)
```
Comment (wshwsh12) [inline expression/builtin_math_vec.go]:
```suggestion
	buf2, err := b.bufAllocator.get(types.ETInt, n)
```

File: expression/builtin_math_vec.go [hunk 2]
```diff
@@ -1049,11 +1049,101 @@ func (b *builtinSignSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) er
...
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf2)
+	buf3, err := b.bufAllocator.get(types.ETString, n)
```
Comment (wshwsh12) [inline expression/builtin_math_vec.go]:
```suggestion
	buf3, err := b.bufAllocator.get(types.ETInt, n)
```

File: expression/builtin_math_vec.go [hunk 3]
```diff
@@ -1049,11 +1049,101 @@ func (b *builtinSignSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) er
...
+	result.ReserveString(n)
+	fromBase := buf2.Int64s()
+	toBase := buf3.Int64s()
+	var (
+		signed     bool
+		negative   bool
+		ignoreSign bool
+	)
+	for i := 0; i < n; i++ {
+		if buf2.IsNull(i) || buf2.IsNull(i) {
```
Comment (wshwsh12) [inline expression/builtin_math_vec.go]:
```suggestion
		if buf1.IsNull(i) || buf2.IsNull(i) || buf3.IsNull(i) {
```

File: expression/builtin_math_vec.go [hunk 4]
```diff
@@ -1049,11 +1049,101 @@ func (b *builtinSignSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) er
...
+		return err
+	}
+	result.ReserveString(n)
+	fromBase := buf2.Int64s()
+	toBase := buf3.Int64s()
+	var (
+		signed     bool
+		negative   bool
+		ignoreSign bool
+	)
```
Comment (wshwsh12) [inline expression/builtin_math_vec.go]: We should reset the boolean values for each loop.

Comment (ZhaoQian888) [reply expression/builtin_math_vec.go]: Reset for each loop can be absolutely right, I forget to consider these situations

File: expression/builtin_math_vec.go [hunk 5]
```diff
@@ -1049,11 +1049,104 @@ func (b *builtinSignSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) er
...
+	for i := 0; i < n; i++ {
+		if buf1.IsNull(i) || buf2.IsNull(i) || buf2.IsNull(i) {
```
Comment (wshwsh12) [inline expression/builtin_math_vec.go]:
```suggestion
		if buf1.IsNull(i) || buf2.IsNull(i) || buf3.IsNull(i) {
```

File: expression/builtin_math_vec.go [hunk 6]
```diff
@@ -1049,11 +1049,104 @@ func (b *builtinSignSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) er
...
+		}
+		if ignoreSign && negative {
+			val = 0 - val
+		}
+		s := strconv.FormatUint(val, int(toBase[i]))
+		if negative && ignoreSign {
+			s = "-" + s
+		}
+		res := strings.ToUpper(s)
+		result.AppendString(res)
```
Comment (qw4990) [inline expression/builtin_math_vec.go]: How about extract this logic to a new method like `(b * builtinConvSig) conv(n string, from, to int) (string, bool, error)` then both the vectorized and row-based implementation can use it.

Comment (ZhaoQian888) [reply expression/builtin_math_vec.go]: That is a good idea.let me do this

**Pull request merged**